kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java

```diff
@@ -353,11 +374,22 @@ private AmazonS3 getConnection(String bucketName) {
     public List<ObjectDto> getAllObjects(String account, String id) {
 
         List<S3ObjectSummary> os = null;
-        String searchPattern=id+SEPARATOR;
-   	   if(useAccountAsBucketname)
-           os = getConnection(account).listObjects(account, searchPattern).getObjectSummaries();
-   	   else
-           os = getConnection(id).listObjects(searchPattern).getObjectSummaries();
+
+   	   if(useAccountAsBucketname) {
+			String searchPattern = id + SEPARATOR;
+			if (prefix != null) {
+				account = prefix + Hyphen + account;
+			}
+			os = getConnection(account).listObjects(account, searchPattern).getObjectSummaries();
+		}
+
+		else {
+			if (prefix != null) {
+				id = prefix + Hyphen + id;
+			}
+			String searchPatternId = id + SEPARATOR;
```

Comment:
Why is this line duplicated and not above the if condition ?

Reply:
Actually for the else condition there should not be any search pattern, since bucket name itself is id. Please create a separate task on this and lets fix this later.

Reply:
as prefix is adding to id then we need to declare searchpattern otherwise it will search for only id without prefix
	id = bucketNamePrefix + id; then String searchPatternId = id + SEPARATOR;

Reply:
I feel we should replace "os = getConnection(id).listObjects(searchPatternId).getObjectSummaries();" to "os = getConnection(id).listObjects(id).getObjectSummaries();"

Reply:
fixed